app/src/androidTest/java/com/github/se/travelpouch/ui/travel/EditTravelScreenTest.kt

```diff
@@ -158,52 +183,90 @@ class EditTravelSettingsScreenTest {
+    inputText("addUserEmailField", "", randomEmail)
 
     // Now this is a valid user that does exist
     doAnswer { invocation ->
           val email = invocation.getArgument<String>(0)
-          val onSuccess = invocation.getArgument<(Profile?) -> Unit>(1)
+          val onSuccess = invocation.getArgument<(fsUid?) -> Unit>(1)
           val customUserInfo =
               Profile(
-                  fsUid = "abcdefghijklmnopqrstuvwxyz12",
+                  fsUid = profileModelView.profile.value.fsUid,
                   name = "Custom User",
                   userTravelList = listOf("00000000000000000000"),
                   email = email,
                   username = "username",
                   friends = null)
           // Call the onSuccess callback with the custom UserInfo
-          onSuccess(customUserInfo)
+          onSuccess(customUserInfo.fsUid)
         }
-        .`when`(travelRepository)
-        .checkParticipantExists(any(), any(), any())
+        .`when`(profileRepository)
+        .getFsUidByEmail(any(), any(), any())
     // Mock the repository.updateTravel method to do nothing
     doNothing().`when`(travelRepository).updateTravel(any(), any(), any())
+    doAnswer { "sigmasigmasigmasigm2" }.`when`(travelRepository).getNewUid()
+    composeTestRule.onNodeWithTag("addUserButton").performClick()
+
+    doAnswer { invocation ->
+          val email = invocation.getArgument<String>(0)
+          val onSuccess = invocation.getArgument<(fsUid?) -> Unit>(1)
+          val customUserInfo =
+              Profile(
+                  fsUid = "sigmasigmasigmasigma12345678",
+                  name = "Custom User",
+                  userTravelList = listOf("00000000000000000000"),
+                  email = email,
+                  username = "username",
+                  friends = null)
+          // Call the onSuccess callback with the custom UserInfo
+          onSuccess(customUserInfo.fsUid)
+        }
+        .`when`(profileRepository)
+        .getFsUidByEmail(any(), any(), any())
+    doAnswer { "sigmasigmasigmasigma" }.`when`(travelRepository).getNewUid()
+    composeTestRule.onNodeWithTag("addUserButton").performClick()
+
+    doAnswer { invocation ->
+          val email = invocation.getArgument<String>(0)
+          val onSuccess = invocation.getArgument<(fsUid?) -> Unit>(1)
+          val customUserInfo =
+              Profile(
+                  fsUid = "abcdefghijklmnopqrstuvwxyz12",
+                  name = "Custom User",
+                  userTravelList = listOf("00000000000000000000"),
+                  email = email,
+                  username = "username",
+                  friends = null)
+          // Call the onSuccess callback with the custom UserInfo
+          onSuccess(customUserInfo.fsUid)
+        }
+        .`when`(profileRepository)
+        .getFsUidByEmail(any(), any(), any())
+    doAnswer { "sigmasigmasigmasigma" }.`when`(travelRepository).getNewUid()
```

Comment:
PS : after reviewing previous code, I noticed that the test was not at fault, but the implementation was. You need to separate the uids of travels from uids of notifications. This is important. 
Also, verify that a call to getNewUid was done, otherwise, the test do not test this part of the code.

Reply:
I have designed a new test that better suits you UI, we can discuss later